### PR TITLE
Add support for `method_exists()` guards

### PR DESCRIPTION
**Pull request by staabm**
implementation for https://github.com/johnbillion/wp-compat/pull/6

---

in case you find more variants which needs fixing and are not covered yet, please send another failing test-case and I will provide a followup

### CODE REVIEW

File: src/Rules/MethodExistsVisitor.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php declare(strict_types = 1);
+
+namespace WPCompat\PHPStan\Rules;
+
+use PhpParser\Node;
+use PhpParser\NodeVisitorAbstract;
+use PHPStan\Node\Printer\ExprPrinter;
+use function count;
+
+final class MethodExistsVisitor extends NodeVisitorAbstract {
```
Comment (staabm) [inline src/Rules/MethodExistsVisitor.php:10]: I tried several variants, even those mentioned in https://github.com/phpstan/phpstan/discussions/11648

I went with the visitor, as I got the impression that this `SinceVersionRule` is expected to work on the `$requiresAtLeast` parameter of `SinceVersionRule`, no matter which version of wordpress itself is installed at the time of analysis.

Comment (johnbillion) [reply src/Rules/MethodExistsVisitor.php:10]: Yep that's correct 👍

**Comment by staabm** [comment]
I think this should work as a initial iteration (in case you don't have remarks)

**Review by johnbillion — APPROVED** [review]
Thank you so much Markus. This is all looking good and everything is passing.

**Pull request merged**